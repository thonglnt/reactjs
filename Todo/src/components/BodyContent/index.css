.content-wrapper {
  display: grid;
  justify-content: center;
}

@media screen and (max-width: 576px) {
  .content-wrapper {
    justify-content: end;
    margin-right: 18%;
  }
}

@media screen and (max-width: 460px) {
  .content-wrapper {
    justify-content: end;
    margin-right: 10%;
  }
}
