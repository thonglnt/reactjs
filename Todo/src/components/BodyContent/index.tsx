// Lib
import { ReactNode } from 'react';

export type BodyContentProps = {
  children: ReactNode | JSX.Element;
};

const BodyContent = ({ children }: BodyContentProps) => {
  return <div className='content-wrapper'>{children}</div>;
};

export default BodyContent;
