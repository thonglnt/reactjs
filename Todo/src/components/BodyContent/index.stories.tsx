// Lib
import { Story } from '@storybook/react';

// Components
import BodyContent, { BodyContentProps } from '@components/BodyContent';

export default {
  title: 'BodyContent',
  component: BodyContent,
};

export const BodyContentSample: Story<BodyContentProps> = () => (
  <BodyContent children={<div>Body content</div>} />
);
