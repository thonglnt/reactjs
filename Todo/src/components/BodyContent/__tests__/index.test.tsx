// Lib
import { render } from '@testing-library/react';

// Components
import BodyContent, { BodyContentProps } from '@components/BodyContent';

describe('Test BodyContent component', () => {
  const initBodyContent: BodyContentProps = {
    children: <></>,
  };

  test('It should render correctly with initial Props', () => {
    const { container } = render(<BodyContent {...initBodyContent} />);

    expect(container).toMatchSnapshot();
  });
});
