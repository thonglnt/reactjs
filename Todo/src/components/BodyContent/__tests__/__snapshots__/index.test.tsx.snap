// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BodyContent component It should render correctly with initial Props 1`] = `
<div>
  <div
    class="content-wrapper"
  />
</div>
`;
